### ISSUE TITLE
Unverified Breakpoint in VSCode. DebugService: Error serving requestsError: FormatException: Unsupported URI form.

### ISSUE THREAD

**Issue by DanielJaramillo94** [flutter/flutter#146795]
### Steps to reproduce

I'm learning to use Flame and I have a little project setted up. Thing is I want to use the debbuger but I can't in my main PC but in the laptop I actually can. 

I have 2 breakpoints:
1. `main.dart`, line 5
2. `pixel_adventure.dart`, line 5

The execution code stops in breakpoint 1, as expected, but it doesn't stop in breakpoint 2. When I do click "Step into" when I'm on breakpoint 1, the debug console prints the next error: `DebugService: Error serving requestsError: FormatException: Unsupported URI form: C:/Users/daniJ/AppData/Local/Pub/Cache/hosted/pub.dev/flame-1.17.0/lib/src/components/sprite_animation_group_component.dart`

The breakpoint 2 is marked as "Unverified endpoint" everytime, as if the execution code wouldn't be able to reach it, but the project is working as expected, so the code is being reached.

### Expected results

I expect the stop execution stops on breakpoints 1 and 2 when running the project in debug mode

### Actual results

The execution stops only in breakpoint 1

### Code sample

_EDITED example to make it shorter_
<details open><summary>Code sample</summary>
Lib folder in project structure (only 3 files):

>_lib (folder)_
>main.dart
>pixel_adventure.dart
>_components (folder)_
>>player.dart

`main.dart`
```dart
import 'package:flutter/material.dart';
import 'package:pixel_adventure/pixel_adventure.dart';

void main() async {
  final game = PixelAdventure();

  runApp(
    Container(),
  );
}
```

`pixel_adventure.dart`
```dart
import 'package:pixel_adventure/components/player.dart';

class PixelAdventure {
  PixelAdventure() {
    final player = Player();
  }
}
```

`player.dart`
```dart
import 'package:flame/components.dart';
import 'package:flutter/services.dart';
import 'package:pixel_adventure/pixel_adventure.dart';

class Player extends SpriteAnimationGroupComponent with KeyboardHandler {
  @override
  bool onKeyEvent(KeyEvent event, Set<LogicalKeyboardKey> keysPressed) {
    return super.onKeyEvent(event, keysPressed);
  }
}
```

</details>

### Logs

<details open><summary>Logs</summary>

```console
DebugService: Error serving requestsError: FormatException: Unsupported URI form: C:/Users/daniJ/AppData/Local/Pub/Cache/hosted/pub.dev/flame-1.17.0/lib/src/components/sprite_animation_group_component.dart
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.19.5, on Microsoft Windows [Version 10.0.22631.2861], locale es-CO)
    • Flutter version 3.19.5 on channel stable at C:\Users\daniJ\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 300451adae (3 weeks ago), 2024-03-27 21:54:07 -0500
    • Engine revision e76c956498
    • Dart version 3.3.3
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\daniJ\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2023.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)

[√] VS Code (version 1.88.1)
    • VS Code at C:\Users\daniJ\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.86.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.2861]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 123.0.6312.123
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 123.0.2420.97

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by DanielJaramillo94** [comment]
Hello @huycozy 
I already raised the issue in Flame repo (https://github.com/flame-engine/flame/issues/3126) but I'm not sure if the bug is on their code, because the bug can dissapear if I just remove one import from a file.

You could check the reproducible example I set on that issue as it is much shorter, or I can paste it in here if needed. The thing is I needed to keep Flame as dependency to raise the error. 😢

**Comment by huycozy** [comment]
I would suggest you follow up on that issue to see if someone there can move it forward. I will keep this issue open, please come back once you have more information.

**Comment by spydon** [comment]
@huycozy I'm fairly confident that this has nothing to do with Flame, since it is Flutter's debugger that is giving the error.
@DanielJaramillo94 could is possibly be a broken local cache or something like that?

**Comment by vmtest888** [comment]
I ran "pub cache clean" and then "pub get" and it was also an error.

**Comment by DanielJaramillo94** [comment]
@spydon It can't be cache. I even reseted my Win11 installation.

**Comment by spydon** [comment]
And this only happens when you have breakpoints in Flame code, not when you have them in for example the counter example?

**Comment by DanielJaramillo94** [comment]
@spydon From the tests I made, happens with some breakpoints in some Flame code. In the provided example with Flame code I have 2 breakpoints and the debugger only fails on break the execution in breakpoint #2.

The error might happen on some code outside Flame code but I couldn't find it.

**Comment by spydon** [comment]
I see that you have android studio installed too, have you tried it in there too?

**Comment by DanielJaramillo94** [comment]
@spydon I have not. I'll do it and share here the results.

**Comment by DanielJaramillo94** [comment]
Well, it works for me as expected in Android Studio. So we know now who to blame 😅

**Comment by DanTup** [comment]
@DanielJaramillo94 I tried to reproduce this using your code sample from above, using Flutter 3.19.5 as included in your Doctor output, I see a compile error:

![image](https://github.com/flutter/flutter/assets/1078012/5992b75d-8aff-460a-af14-0f2ee931d05d)

This seems to be a breaking change that was made in Flame 1.16.0 so it seems like the sample code may have been written against an older version of Flame - however your error message mentions flame 1.17.0.

Can you provide a copy of your repro project _including pubspec.lock_ so we can try to reproduce with exactly the same code and dependency versions?

Could you also provide a screenshot of where you are seeing this error? It's not currently clear to me if this error is coming from (and searching the web for the strings "Error serving requests" and "Unsupported URI form" hasn't turned up much to help me understand what might be the source of those errors). It appears like something is trying to parse a file path (with the wrong direction slashes) as a URI, but it's not clear to me where that path has come from.

**Comment by DanielJaramillo94** [comment]
@DanTup sure!
Sorry for providing a code that not compiles, I was trying with another version of Flutter (the breakpoint fails to be reached anyways) and forgot to change the code after to a flutter 3.19.5 style version

Here is [the repo](https://github.com/DanielJaramillo94/flutter-flame-breakpoint-fails) containing the code that fails compiling with Flutter 3.19.5.

The error message only appears if I hit _Step Into_ button while I'm paused in breakpoint 1. If I hit _Continue_ button, the error is not shown in console.
![image](https://github.com/flutter/flutter/assets/29381519/ab782e47-f04a-4f22-ac9e-7bd6df77f1bb)

**Comment by huycozy** [comment]
@DanielJaramillo94 I pulled your repo above and started debugging it. There are no corresponding lines as mentioning from the beginning so I set two breakpoints at `Container()` in `main.dart` and `final player = Player()` in `pixel_adventure.dart` files; then click on `Step Into` by the code flow; but I couldn't reproduce the error (app ran normally in the end of debugging steps)

Maybe I put breakpoints on wrong lines. Could you update the steps?

Also, could you share Flutter and Dart extensions version on your VSCode? (For me, they are Flutter and Dart `v3.88.0` on VSCode `Version: 1.89.0`).

**Comment by DanielJaramillo94** [comment]
@huycozy I updated the breakpoints:
1. `main.dart`, line 5
2. `pixel_adventure.dart`, line 5

As a note, it doesn't matter where I set the breakpoint in `main.dart` file, it always fires the breakpoint; and it doesn't matter where I set the breakpoint in `pixel_adventure.dart` file, it never fires. It like the debug service cannot reach `pixel_adventure.dart` file.

I would also clarify that the error occurs no matter if I click on _Continue_ or _Step Into_ buttons while the debug is paused in breakpoint 1. The only difference is that **the error message** only appears in _Debug console_ if I click on _Step Into_ button.

### flutter doctor -v

<details open>

```console
[√] Flutter (Channel stable, 3.19.5, on Microsoft Windows [Version 10.0.22631.3447], locale es-CO)
    • Flutter version 3.19.5 on channel stable at C:\Users\daniJ\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 300451adae (6 weeks ago), 2024-03-27 21:54:07 -0500
    • Engine revision e76c956498
    • Dart version 3.3.3
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\daniJ\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2023.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)

[√] VS Code (version 1.89.1)
    • VS Code at C:\Users\daniJ\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.88.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.3447]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 124.0.6367.201
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 124.0.2478.80

[√] Network resources
    • All expected network resources are available.
```

</details>

**Comment by huycozy** [comment]
I updated my VSCode to be the same as yours (1.89.1) but couldn't reproduce it. All debug steps are going well. Could you check the recorded video below to see if I missed a step?

<details>
<summary>Demo</summary>

https://github.com/flutter/flutter/assets/104349824/e0c7c591-8e8d-4e2d-8117-43d738fb6759

</details>

**Comment by DanTup** [comment]
@DanielJaramillo94 I was also unable to reproduce the issue. Is it possible you could capture a log for me?

- Open the same sample project you shared above (so I can use the same code when debugging)
- Run the [**Dart: Capture Debugging Logs** command](https://dartcode.org/docs/logging/#capture-debugging-logs)
- Start running the project
- Reproduce the issue so you see the error in the Debug Console
- Click **Cancel** on the logging notification to stop logging and open the log
- Remove anything sensitive from the log
- Attach the log here (or in a [gist](https://gist.github.com/))

Note: The log may be quite large and can contain parts of your source code (and file paths from your machine). It includes communication between VS Code the the Dart VM.

Thanks!

**Comment by DanielJaramillo94** [comment]
@huycozy I see those steps right. However, I would like to say:

1. It is just failing when I debug in Chrome
2. I myself am unable to reproduce the bug in my laptop. It just fails on my desktop. This is why I thought it was a local problem and I did format the PC

@DanTup here are [the logs](https://gist.github.com/DanielJaramillo94/d8591b5b2efc10e87338b54e71f6580a)

Thank you both :D

**Comment by DanTup** [comment]
@DanielJaramillo94 Thanks for the log. I was unable to reproduce the issue still (I didn't realise you were using Chrome earlier, but I retried there and still could not reproduce). However in the logs I saw a slightly different error that included:

```
[3:12:12 PM] [General] [Info] [Flutter (Chrome)] <== [Flutter] [  +36 ms] DevHandler: VmService proxy responded with an error:
[3:12:12 PM] [General] [Info] [Flutter (Chrome)] <== [Flutter]            {jsonrpc: 2.0, id: 57, error: {code: -32603, message: Unexpected DWDS error for addBreakpointWithScriptUri: FormatException: Unsupported URI form: C:/Users/daniJ/AppData/Local/Pub/Cache/hosted/pub.dev/flame-1.17.0/lib/src/components/sprite_animation_group_component.dart}}
[3:12:12 PM] [General] [Info] [Flutter (Chrome)] <== [VM] {"jsonrpc":"2.0","error":{"code":-32603,"message":"Unexpected DWDS error for addBreakpointWithScriptUri: FormatException: Unsupported URI form: C:/Users/daniJ/AppData/Local/Pub/Cache/hosted/pub.dev/flame-1.17.0/lib/src/components/sprite_animation_group_component.dart","data":{"request":{"jsonrpc":"2.0","id":"42","method":"addBreakpointWithScriptUri","params":{"isolateId":"1","scriptUri":"file:///D:/Documents/Study/Flutter/Bugs/flutter-flame-breakpoint-fails/lib/pixel_adventure.dart","line":5}}}},"id":"42"}
```

> Unexpected DWDS error for addBreakpointWithScriptUri: FormatException: Unsupported URI form: C:/Users/daniJ/AppData/Local/Pub/Cache/hosted/pub.dev/flame-1.17.0/lib/src/components/sprite_animation_group_component.dart

Interestingly, that's the same error I reported in https://github.com/dart-lang/webdev/issues/2396 however I only saw that issue when I had the macros experiment enabled. Can you confirm whether you've tried using the macros experiment on this project?

If not, perhaps @elliette may have some ideas about whether the same error could occur in other situations (without macros enabled)? (the PR that fixed that issue does mention macros, although I don't know the code well enough to understand if it's more general than that).

As noted above, there has been a new release of Flutter - it's worth re-testing with that, as there will have been many bugfixes (including a newer version of DWDS) that might have affected things here.

**Comment by elliette** [comment]
Ah yes unfortunately this error wasn't limited to projects with `macros`. The fix is in DWDS 24.0.0, which is currently on the `master` branch of Flutter (rolled in as part of https://github.com/flutter/flutter/pull/147546), but isn't on the `beta` or `stable` branches yet.

**Comment by DanielJaramillo94** [comment]
I guess this is not already necessary as @elliette states, but the error stills appears in Flutter 3.22 with no use of macros. I switched to master branch (3.22.0-28.0.pre.3) and the breakpoints now work as expected!
